pdns/arguments.cc

```diff
@@ -460,43 +460,35 @@ bool ArgvMap::file(const char *fname, bool lax, bool included)
 
 void ArgvMap::gatherIncludes(std::vector<std::string> &extraConfigs) {
   extraConfigs.clear();
-  if (params["include-dir"].empty()) return; // nothing to do
-    struct stat st;
-    DIR *dir;
-    struct dirent *ent;
-
-    // stat
-    if (stat(params["include-dir"].c_str(), &st)) {
-       g_log << Logger::Error << params["include-dir"] << " does not exist!" << std::endl;
-       throw ArgException(params["include-dir"] + " does not exist!");
-    }
-
-    // wonder if it's accessible directory
-    if (!S_ISDIR(st.st_mode)) {
-       g_log << Logger::Error << params["include-dir"] << " is not a directory" << std::endl;
-       throw ArgException(params["include-dir"] + " is not a directory");
-    }
-
-    if (!(dir = opendir(params["include-dir"].c_str()))) {
-       g_log << Logger::Error << params["include-dir"] << " is not accessible" << std::endl;
-       throw ArgException(params["include-dir"] + " is not accessible");
-    }
+  if (params["include-dir"].empty())
+    return; // nothing to do
+
+  DIR *dir;
+  if (!(dir = opendir(params["include-dir"].c_str()))) {
+    int err = errno;
+    string msg = params["include-dir"] + " is not accessible: " + strerror(err);
+    g_log << Logger::Error << msg << std::endl;
+    throw ArgException(msg);
```

Comment:
Not a big deal since I believe we will exit right away, but this would leak the `DIR*`.

Reply:
I don't think so since the opendir() call failed...

Reply:
Oh, right, sorry about that!